src/button/pay.js

```
@@ -114,8 +112,35 @@ export function initiatePaymentFlow({ payment, serviceData, config, components,
             const validateOrderPromise = createOrder().then(orderID => {
                 return validateOrder(orderID, { env, clientID, merchantID, intent, currency, vault });
             });
-
-            const confirmOrderPromise = smartFields && smartFields.confirm && createOrder().then(smartFields.confirm);
+            
+            const confirmOrderPromise = createOrder()
+                .then((orderID) => {
+                    return window.xprops.sessionState.get(
+                        `__confirm_${ fundingSource }_payload__`
+                    )
+                        .then(confirmOrderPayloadSessionStorage => {
+                            if (!confirmOrderPayloadSessionStorage) {
+                                // noop when there is no confirm call to make (ie. regular flow).
+                                return;
+                            }
+
+                            let confirmOrderPayload;
+                            
+                            try {
+                                confirmOrderPayload = JSON.parse(confirmOrderPayloadSessionStorage);
+                            } catch (err) {
```

Do we want to re-throw the error here? If the data isn't available the buyer shouldn't be able to check out right?